core/src/main/java/org/apache/carbondata/core/scan/result/impl/NonFilterQueryScannedResult.java

```diff
@@ -48,6 +53,31 @@ public NonFilterQueryScannedResult(BlockExecutionInfo blockExecutionInfo) {
     return getDictionaryKeyIntegerArray(currentRow);
   }
 
+  @Override public List<byte[]> getDictionaryKeyArrayBatch(int batchSize) {
+    ++currentRow;
```

Comment:
Is this increment is intentional? Because increment as per batch size is already happening in `RawBasedResultCollector`